src/domain/UBGraphicsTextItem.cpp

```diff
@@ -139,6 +139,44 @@ QFont UBGraphicsTextItem::createDefaultFont()
     return font;
 }
 
+bool UBGraphicsTextItem::event(QEvent *ev)
```

Comment:
Can't you directly modify `KeyPressEvent` instead of checking every event to see if it's a KeyEvent ?